gateway/manager/watcher/watcher.go

```diff
@@ -41,115 +31,86 @@ func NewFileWatcher(
-		case err, ok := <-s.watcher.Errors:
-			if !ok {
-				return
-			}
-			s.log.Error().Err(err).Msg("Error watching files")
-			sentry.CaptureError(err, nil)
-		}
-	}
-}
-
-// Close closes the file watcher
-func (s *FileWatcher) Close() error {
-	return s.watcher.Close()
-}
-
-func (s *FileWatcher) handleEvent(event fsnotify.Event) {
-	s.log.Info().Str("event", event.String()).Msg("File event")
-
-	filename := filepath.Base(event.Name)
-	switch event.Op {
-	case fsnotify.Create:
-		s.OnFileChanged(filename)
-	case fsnotify.Write:
-		s.OnFileChanged(filename)
-	case fsnotify.Rename:
-		s.OnFileDeleted(filename)
-	case fsnotify.Remove:
-		s.OnFileDeleted(filename)
-	default:
-		err := ErrUnknownFileEvent
-		s.log.Error().Err(err).Str("filename", filename).Msg("Unknown file event")
-		sentry.CaptureError(sentry.SentryError(err), nil, sentry.Extras{"filename": filename, "event": event.String()})
+// OnFileChanged implements watcher.FileEventHandler
+func (s *FileWatcher) OnFileChanged(filePath string) {
+	// Check if this is actually a file (not a directory)
+	if info, err := os.Stat(filePath); err != nil || info.IsDir() {
+		return
 	}
-}
-
-func (s *FileWatcher) OnFileChanged(filename string) {
-	// Construct full file path
-	filePath := filepath.Join(s.watchPath, filename)
 
 	// Delegate to cluster registry
 	if err := s.clusterRegistry.UpdateCluster(filePath); err != nil {
-		s.log.Error().Err(err).Str("filename", filename).Str("path", filePath).Msg("Failed to update cluster")
-		sentry.CaptureError(err, sentry.Tags{"filename": filename})
+		s.log.Error().Err(err).Str("path", filePath).Msg("Failed to update cluster")
+		sentry.CaptureError(err, sentry.Tags{"filepath": filePath})
 		return
 	}
 
-	s.log.Info().Str("filename", filename).Msg("Successfully updated cluster from file change")
+	s.log.Info().Str("path", filePath).Msg("Successfully updated cluster from file change")
 }
 
-func (s *FileWatcher) OnFileDeleted(filename string) {
-	// Construct full file path
-	filePath := filepath.Join(s.watchPath, filename)
-
+// OnFileDeleted implements watcher.FileEventHandler
+func (s *FileWatcher) OnFileDeleted(filePath string) {
 	// Delegate to cluster registry
 	if err := s.clusterRegistry.RemoveCluster(filePath); err != nil {
-		s.log.Error().Err(err).Str("filename", filename).Str("path", filePath).Msg("Failed to remove cluster")
-		sentry.CaptureError(err, sentry.Tags{"filename": filename})
+		s.log.Error().Err(err).Str("path", filePath).Msg("Failed to remove cluster")
+		sentry.CaptureError(err, sentry.Tags{"filepath": filePath})
 		return
 	}
 
-	s.log.Info().Str("filename", filename).Msg("Successfully removed cluster from file deletion")
+	s.log.Info().Str("path", filePath).Msg("Successfully removed cluster from file deletion")
+}
+
+// loadAllFiles loads all files in the directory and subdirectories
+func (s *FileWatcher) loadAllFiles(dir string) error {
+	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
The number of files is probably low otherwise I would suggest to use `filepath.WalkDir()` as it is faster if you don't need file stats.